### ISSUE TITLE
Request method in Access Rules

### ISSUE THREAD

**Issue by r0man** [funcool/buddy-auth#18]
Hello,

I have a question/feature request. Say I have a rule defined like
this:

```
{:uri "/comments" :handler authenticated-user}
```

This would match both `GET /comments` and `POST /comments`. What
I really want is just match `POST /comments`. I could do this
with the following rule:

```
{:uri "/comments" :handler {:or [get-request authenticated-user]}}}
```

It feels a bit strange to have to "unmatch" a rule when I want to
match on the request method as well. I think the common case is
that you have different rules depending on the request method.

Would you be open to add something like this to a rule:

```
{:uri "/comments" :methods #{:post}} :handler authenticated-user}
{:uri "/comments" :methods #{:delete}} :handler authenticated-user}
```

This would match only ŕequests with the given request
methods. Without the :methods keyword it would match everything
as is. I think this would make writing access rules a bit easier.

What do you think?

Roman

**Comment by niwinz** [comment]
Seems a very great Idea! 

Are you thinking implement it or is a simple feature request?

**Comment by r0man** [comment]
@niwibe I'll give it a try and send a PR later. Ok?

**Comment by niwinz** [comment]
Great \o/

**Comment by niwinz** [comment]
Fixed with #19
